Patch-Source: https://github.com/dunst-project/dunst/commit/62be5b46c5ee69ce675e12a86d73c81d324a13b1
---
From 62be5b46c5ee69ce675e12a86d73c81d324a13b1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20=C4=8Cerm=C3=A1k?= <dcermak@suse.com>
Date: Fri, 20 Dec 2024 10:43:01 +0100
Subject: [PATCH] Change type of bool to gboolean

`bool` and `gboolean` are not exactly the same type, they only evaluate to the
same value in C on little endian architectures but evaluate differently on big
endian. This causes issues on e.g. s390x as described in #1421

This fixes https://github.com/dunst-project/dunst/issues/1421

Co-authored-by: Josef Reidinger <jreidinger@suse.com>
Co-authored-by: Aleksei Bavshin <alebastr@fedoraproject.org>
---
 src/rules.h | 2 +-
 test/dbus.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/rules.h b/src/rules.h
index bdeb17004..9f99c01de 100644
--- a/src/rules.h
+++ b/src/rules.h
@@ -56,7 +56,7 @@ struct rule {
         char *format;
         char *script;
         enum behavior_fullscreen fullscreen;
-        bool enabled;
+        gboolean enabled;
         int progress_bar_alignment;
         char *set_stack_tag; // this has to be the last modifying rule
 };
diff --git a/test/dbus.c b/test/dbus.c
index 77d7d42bb..66e9e9fdc 100644
--- a/test/dbus.c
+++ b/test/dbus.c
@@ -639,7 +639,7 @@ TEST test_dbus_cb_dunst_RuleList(void)
         g_variant_dict_init(&d, dict);
 
         char *str;
-        bool boolean;
+        gboolean boolean;
 
         ASSERT(g_variant_dict_lookup(&d, "name", "s", &str));
         ASSERT_STR_EQ("testing RuleList", str);
